homeassistant/components/generic/config_flow.py

```diff
@@ -264,9 +283,20 @@ async def async_step_user(
                 if not errors:
                     user_input[CONF_CONTENT_TYPE] = still_format
                     user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE] = False
-                    await self.async_set_unique_id(self.flow_id)
-                    return self.async_create_entry(
-                        title=name, data={}, options=user_input
+                    if user_input.get(CONF_STILL_IMAGE_URL):
+                        await self.async_set_unique_id(self.flow_id)
+                        return self.async_create_entry(
+                            title=name, data={}, options=user_input
+                        )
+                    # If user didn't specify a still image URL,
+                    # we can't (yet) autodetect it from the stream.
+                    # Show a conditional 2nd page to ask them the content type.
+                    self.cached_user_input = user_input
+                    self.cached_title = name
+                    return self.async_show_form(
+                        step_id="user2",
+                        data_schema=build_schema2({}),
+                        errors={},
```

Comment:
Just call the step without user input, and let the step then generate the form if no input is given.

```suggestion
                    return await self.async_step_user2(
```

Reply:
Ok done. 92bb1cb7